pygsti/algorithms/fiducialselection.py

```diff
@@ -185,6 +244,16 @@ def find_fiducials(target_model, omit_identity=True, eq_thresh=1e-6,
             printer.log('Preparation fiducials:', 1)
             printer.log(str([fid.str for fid in prepFidList]), 1)
             printer.log('Score: {}'.format(prepScore.minor), 1)
+            #if requested do a final check with test_fiducial_list
+            #to verify the algorithm succeeds
+            if final_test:
+                final_test_fiducial_list = test_fiducial_list(target_model, prepFidList, 'prep',
+                                                    score_func=algorithm_kwargs['score_func'], return_all=False,
+                                                    threshold=algorithm_kwargs.get('threshold', 1e-6), fid_cache=prep_cache)
+                if final_test_fiducial_list:
+                    printer.log('Final test of the candidate prep fiducial lists passed.', 1)
+                else:
+                    printer.log('Final test of the candidate prep fiducial lists failed.', 1)
```

Comment:
The above block is repeated 8 (I think) times within this function, which makes me think it'd be worth consolidating it to a nested function.

Reply:
I refactored this as suggested.